android/jni/com/mapswithme/maps/editor/Editor.cpp

```diff
@@ -274,4 +287,46 @@ Java_com_mapswithme_maps_editor_Editor_nativeGetNewFeatureCategories(JNIEnv * en
 
   return jCategories;
 }
+
+JNIEXPORT jobjectArray JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeGetCuisines(JNIEnv * env, jclass clazz)
+{
+  osm::TAllCuisines const & cuisines = osm::Cuisines::Instance().AllSupportedCuisines();
+  vector<string> keys(cuisines.size());
+  for (size_t i = 0; i < cuisines.size(); i++)
+    keys[i] = cuisines[i].first;
+  return ToJavaArray(env, keys);
+}
+
+JNIEXPORT jobjectArray JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeGetSelectedCuisines(JNIEnv * env, jclass clazz)
+{
+  return ToJavaArray(env, g_editableMapObject.GetCuisines());
+}
+
+JNIEXPORT jobjectArray JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeGetCuisinesTranslations(JNIEnv * env, jclass clazz)
```

Comment:
Если что, то переводы можно получать только когда необходимо, для нужного ключа, там специальный метод Translate есть.

> On Mar 22, 2016, at 11:19, Dmitry Yunitsky <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/editor/Editor.cpp <https://github.com/mapsme/omim/pull/2391#discussion_r56949459>:
> 
> > +{
> > +  osm::TAllCuisines const & cuisines = osm::Cuisines::Instance().AllSupportedCuisines();
> > +  vector<string> keys(cuisines.size());
> > +  for (size_t i = 0; i < cuisines.size(); i++)
> > +    keys[i] = cuisines[i].first;
> > +  return ToJavaArray(env, keys);
> > +}
> > +
> > +JNIEXPORT jobjectArray JNICALL
> > +Java_com_mapswithme_maps_editor_Editor_nativeGetSelectedCuisines(JNIEnv * env, jclass clazz)
> > +{
> > +  return ToJavaArray(env, g_editableMapObject.GetCuisines());
> > +}
> > +
> > +JNIEXPORT jobjectArray JNICALL
> > +Java_com_mapswithme_maps_editor_Editor_nativeGetCuisinesTranslations(JNIEnv * env, jclass clazz)
> Можно, но придётся больше кода писать с конструктором пар, а разницы особо не ощутим больше ни в чём
> … <x-msg://262/#>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> 
>  <https://github.com/mapsme/omim/pull/2391#discussion_r56949237> <https://github.com/mapsme/omim/pull/2391/files/f423aeb4c2632729a07c8764af99345e0b13e77b#r56949237>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2391/files/f423aeb4c2632729a07c8764af99345e0b13e77b#r56949459>